### PR TITLE
🎨 Fix/liquidator

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/Liquidator.sol [hunk 1]
```diff
@@ -14,63 +14,54 @@ import {ILendingPool} from "./interfaces/ILendingPool.sol";
...
+
+    // Fee paid to the Liquidation Initiator.
+    // Defined as a fraction of the openDebt with 2 decimals precision.
+    // Absolute fee can be further capped to a max amount by the creditor.
+    uint8 public initiatorRewardWeight;
+    // Penalty the Vault owner has to pay to the trusted Creditor on top of the open Debt for being liquidated.
+    // Defined as a fraction of the openDebt with 2 decimals precision
+    uint8 public penaltyWeight;
+
+    address public immutable factory;
```
Comment (j-vp) [inline src/Liquidator.sol:44]: move up as first

File: src/Liquidator.sol [hunk 2]
```diff
@@ -14,63 +14,54 @@ import {ILendingPool} from "./interfaces/ILendingPool.sol";
...
-    ///////////////////////////////////////////////////////////////*/
-
-    /**
-     * @notice Sets the factory address on the liquidator.
-     * @param factory_ the factory address.
-     */
-    function setFactory(address factory_) external onlyOwner {
-        factory = factory_;
+        minPriceMultiplier = 50;
+        cutoffTime = 14_400; //4 hours
```
Comment (j-vp) [inline src/Liquidator.sol:64]: add base

File: src/Liquidator.sol [hunk 3]
```diff
@@ -79,41 +70,48 @@ contract Liquidator is Owned {
...
+        require(halfLifeTime < 28_800, "LQ_SACP: halfLifeTime too high"); // 8 hours
+        require(cutoffTime_ > 3_600, "LQ_SACP: cutoff too low"); // 1 hour
+        require(cutoffTime_ < 64_800, "LQ_SACP: cutoff too high"); // 18 hours
+
+        //Derive base from the halfLifeTime
+        uint64 base_ = uint64(1e18 * 1e18 / LogExpMath.pow(2 * 1e18, 1e18 / halfLifeTime));
+
+        //Check that LogExpMath.pow(base, timePassed) does not error at cutoffTime (due to numbers smaller than minimum precision)
+        //Since LogExpMath.pow is a strictly decreasing function checking the power function at cutoffTime
+        //guarantees that the function does not revert on all timestapms between start of the auction and the cutoffTime
```
Comment (j-vp) [inline src/Liquidator.sol:109]:
```suggestion
        //guarantees that the function does not revert on all timestamps between start of the auction and the cutoffTime
```

File: src/Liquidator.sol [hunk 4]
```diff
@@ -179,31 +190,45 @@ contract Liquidator is Owned {
...
      * @return price The total price for which the vault can be purchased.
-     * @dev We use a dutch auction: price constantly decreases and the first bidder buys the vault
-     * And immediately ends the auction.
-     * @dev Price decreases exponentially: P(t) = P(O) * DR^t with P(O) = openDebt * startPriceMultiplier.
+     * @dev We use a dutch auction: price constantly decreases and the first bidder buys the vault and immediately ends the auction
+     * @dev Price P(t) decreases exponentially over time: P(t) = openDebt * [(SPM - MPM) * base^t + MPM]
+     * SPM: The startPriceMultiplier defines the initial price: P(0) = openDebt * SPM (2 decimals precision)
+     * MPM: The minPriceMultiplier defines the assymptotic end price for P(∞) = openDebt * MPM (2 decimals precision)
+     * base: defines how fast the exponential curve decreases (18 decimals precision)
+     * t: time passed since start auction (in seconds, 18 decimals precision)
```
Comment (j-vp) [inline src/Liquidator.sol:206]: add comment: unchecked since library is made when unchecked was the default

**Review by zekiblue — APPROVED** [review]

**Pull request merged**